### PR TITLE
Java Solution for Cellular Network (Implementation 2, Two Pointers)

### PR DESCRIPTION
**Pull request by brad182**
I didn't solve the problem this way, so I don't think I can improve the comments that much

### CODE REVIEW

**Review by maggie-j-liu — APPROVED** [review]

File: solutions/cf-702C.mdx [hunk 1]
```diff
@@ -188,6 +188,74 @@ int main() {
 
 </CPPSection>
 	
+<JavaSection>
+	
+```java
+import java.io.*;
+import java.util.Arrays;
+import java.util.StringTokenizer;
```
Comment (nathangong) [inline solutions/cf-702C.mdx:196]:
```suggestion
import java.util.*;
```

File: solutions/cf-702C.mdx [hunk 2]
```diff
@@ -188,6 +188,73 @@ int main() {
...
+		int n = io.nextInt();
+		int m = io.nextInt();
+		int[] dist = new int[MAXIMUM_SIZE];
+
+		Arrays.fill(dist, 2 * MOD);
+
+		int[] a = new int[n];
+		int[] b = new int[m];
+
+		for (int x = 0; x < n; x++) {
```
Comment (nathangong) [inline solutions/cf-702C.mdx:213]: Why not i

File: solutions/cf-702C.mdx [hunk 3]
```diff
@@ -188,6 +188,73 @@ int main() {
...
+			}
+			dist[i] = Math.min(dist[i], Math.abs(a[i] - b[pointer1]));
+		}
+
+		int answer = 0;
+		// the maximum in the array is the answer
+		for (int i = 0; i < n; i++) {
+			answer = Math.max(answer, dist[i]);
+		}
+
```
Comment (SansPapyrus683) [inline solutions/cf-702C.mdx:248]: 207, 212, 216, 222, 231, 233, and 248 aren't necessary

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

**Review by HarryWangATX — APPROVED** [review]

File: solutions/cf-702C.mdx (hunk 2 above)
Comment (brad182) [reply solutions/cf-702C.mdx:213]: bro

File: solutions/cf-702C.mdx [hunk 4]
```diff
@@ -188,6 +188,73 @@ int main() {
...
 	
+<JavaSection>
+	
+```java
+import java.io.*;
+import java.util.*;
+
+public class CellularNetwork2 {
+	static final int MOD = 1000000007;
+	static final int MAXIMUM_SIZE = 100000;
```
Comment (SansPapyrus683) [inline solutions/cf-702C.mdx:199]: just MAX_SIZE is ok
but this is alright

File: solutions/cf-702C.mdx (hunk 3 above)
Comment (SansPapyrus683) [reply solutions/cf-702C.mdx:248]: not addressed

File: solutions/cf-702C.mdx [hunk 5]
```diff
@@ -188,6 +188,73 @@ int main() {
...
+			while (pointer1 < (m - 1) && b[pointer1 + 1] <= a[i]) {
+				// keep on incrementing
+				pointer1++;
+			}
+			dist[i] = Math.min(dist[i], Math.abs(a[i] - b[pointer1]));
+		}
+
+		pointer1 = m - 1;
+
+		// pointer going right to left
```
Comment (SansPapyrus683) [inline solutions/cf-702C.mdx:234]: these comments don't really explain anything tbh
i mean what does the pointer going to the left *do*

File: solutions/cf-702C.mdx (hunk 2 above)
Comment (nathangong) [reply solutions/cf-702C.mdx:213]: what

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: solutions/cf-702C.mdx (hunk 2 above)
Comment (nathangong) [reply solutions/cf-702C.mdx:213]: you used i as the for loop variable everywhere else

File: solutions/cf-702C.mdx (hunk 3 above)
Comment (brad182) [reply solutions/cf-702C.mdx:248]: its for spacing its an optional change harry thinks that its fine

Comment (SansPapyrus683) [reply solutions/cf-702C.mdx:248]: alr fine

File: solutions/cf-702C.mdx (hunk 2 above)
Comment (SansPapyrus683) [reply solutions/cf-702C.mdx:213]: yeah be consistent lol

File: solutions/cf-702C.mdx (hunk 5 above)
Comment (brad182) [reply solutions/cf-702C.mdx:234]: i don't know, if i did every problem in a way where I tried to understand everything, adding support to java to every problem would take forever

Comment (nathangong) [reply solutions/cf-702C.mdx:234]: Well you probably want to try to make sure you understand the problem so you can help people who are trying to read through your code understand what everything is doing. Imo quality > quantity

File: solutions/cf-702C.mdx [hunk 6]
```diff
@@ -188,6 +188,73 @@ int main() {
...
+		for (int i = n - 1; i >= 0; i--) {
+			while (pointer1 > 0 && b[pointer1 - 1] >= a[i]) {
+				// keep on decrementing
```
Comment (ryanchou-dev) [inline solutions/cf-702C.mdx:237]:
```suggestion
```
don't think this is really necessary

File: solutions/cf-702C.mdx [hunk 7]
```diff
@@ -188,6 +188,73 @@ int main() {
...
+			answer = Math.max(answer, dist[i]);
+		}
+
+		io.println(answer);
+		io.close();
+	}
+	//CodeSnip{Kattio}
+}	
+```
+
```
Comment (ryanchou-dev) [inline solutions/cf-702C.mdx:255]:
```suggestion
```

**Review by ryanchou-dev — APPROVED** [review]
nice (changes below are optional)

**Review by dongliu0426 — APPROVED** [review]
nice

**Pull request merged**